src/Evolution/Systems/Cce/AnalyticSolutions/GaugeWave.hpp

```diff
@@ -0,0 +1,266 @@
+ private:
+  double coordinate_wave_function(double time) const noexcept;
+
+  double du_coordinate_wave_function(double time) const noexcept;
+
+  double du_du_coordinate_wave_function(double time) const noexcept;
+
+ protected:
+  /// A no-op as the gauge wave solution does not have substantial
+  /// shared computation to prepare before the separate component calculations.
+  void prepare_solution(const size_t /*output_l_max*/,
+                        const double /*time*/) const noexcept override {}
+
+  /*!
+   * \brief Compute the spherical coordinate metric from the closed-form gauge
+   * wave metric.
+   *
+   * \details The transformation of the ingoing Eddington-Finkelstein coordinate
+   * produces metric components in spherical coordinates:
+   *
+   * \f{align*}{
+   * g_{tt} &= \frac{-1}{r^3}\left(r - 2 M\right)
+   * \left(r + \partial_u F(u)\right)^2\\
+   * g_{rt} &= \frac{1}{r^4} \left(r + \partial_u F(u)\right)
+   * \left(2 M r^2 + \left(r - 2 M\right)
+   * \left(r \partial_u F(u) + F(u)\right)\right) \\
+   * g_{rr} &= \frac{1}{r^5} \left(r^2 - r \partial_u F(u) - F(u)\right)
+   * \left(r^3 + 2 M r^2 + \left(r - 2 M\right)
+   * \left(r \partial_u F(u) + F(u)\right)\right) \\
+   * g_{\theta \theta} &= r^2 \\
+   * g_{\phi \phi} &= r^2 \sin^2(\theta),
+   * \f}
+   *
+   * and all other components vanish. Here, \f$F(u)\f$ is defined as
+   *
+   * \f{align*}{
+   * F(u) &= A \sin(\omega u) e^{-(u - u_0)^2 /k^2},\\
+   * \partial_u F(u) &= A \left(-2 \frac{u - u_0}{k^2} \sin(\omega u)
+   * + \omega \cos(\omega u)\right) e^{-(u - u_0)^2 / k^2}.
+   * \f}
+   *
+   * \note The \f$\phi\f$ components are returned in a form for which the
+   * \f$\sin(\theta)\f$ factors are omitted, assuming that derivatives and
+   * Jacobians will be applied similarly omitting those factors (and therefore
+   * improving precision of the tensor expression). If you require the
+   * \f$\sin(\theta)\f$ factors, be sure to put them in by hand in the calling
+   * code.
+   */
+  void spherical_metric(
+      gsl::not_null<
+          tnsr::aa<DataVector, 3, ::Frame::Spherical<::Frame::Inertial>>*>
+          spherical_metric,
+      size_t l_max, double time) const noexcept override;
+
+  /*!
+   * \brief Compute the radial derivative of the spherical coordinate metric
+   * from the closed-form gauge wave metric.
+   *
+   * \details The transformation of the ingoing Eddington-Finkelstein coordinate
+   * produces the radial derivative of the metric components in spherical
+   * coordinates:
+   *
+   * \f{align*}{
+   * \partial_r g_{tt} + \partial_t g_{tt} =& \frac{2}{r^4}
+   * \left(r + \partial_u F(u)\right)
+   * \left(- M r + (r - 3 M)\partial_u F(u)\right) \\
+   * \partial_r g_{rt} + \partial_t g_{rt} =& - \frac{1}{r^5}
+   * \left(2 M r^3 + 2 r F(u) (r - 3M) + \partial_u F(u)(r^3 + F(u)(3r - 8 M)) +
+   * 2 r (\partial_u F(u))^2 (r - 3M)\right)\\
+   * \partial_r g_{rr} + \partial_t g_{rr} =& \frac{2}{r^6}
```

Comment:
I will kindly ask you change parentheses->brackets->braces as you get deeper and deeper. It's very difficult to line up which closing paren goes with which opening paren.